src/common/store-update-message-hub.ts

```diff
@@ -22,15 +22,16 @@ export class StoreUpdateMessageHub {
         this.registeredUpdateListeners[storeId] = listener;
     }
 
-    public readonly handleMessage = (message: StoreUpdateMessage<any>): void => {
+    public readonly handleMessage = (message: StoreUpdateMessage<any>): Promise<void> => {
         if (!this.isValidMessage(message)) {
-            return;
+            return Promise.resolve();
         }
 
         const listener = this.registeredUpdateListeners[message.storeId];
         if (listener) {
             listener(message);
```

Comment:
Particularly, this path through a fire-and-forget `listener` *must not* return `Promise.resolve()`; it needs to be triggering the fire-and-forget delay for now.